Plugins/SwiftLintPlugin/SwiftLintPlugin.swift

```diff
@@ -3,81 +3,146 @@ import PackagePlugin
+                        pluginWorkDirectory: context.pluginWorkDirectory)
     }
 
-    private func createBuildCommands(
-        inputFiles: [Path],
-        packageDirectory: Path,
-        workingDirectory: Path,
-        tool: PluginContext.Tool
-    ) -> [Command] {
-        if inputFiles.isEmpty {
-            // Don't lint anything if there are no Swift source files in this target
-            return []
-        }
+    /// Collects the paths of the Swift files to be linted.
+    private func swiftFiles(target: SourceModuleTarget) -> [Path] {
+        target
+            .sourceFiles(withSuffix: "swift")
+            .map(\.path)
+    }
 
-        var arguments = [
+    /// Creates an environment dictionary containing a value for the `BUILD_WORKSPACE_DIRECTORY` key.
+    ///
+    /// This method locates the topmost `.swiftlint.yml` config file within the package directory for this target
+    /// and sets the config file's containing directory as the `BUILD_WORKSPACE_DIRECTORY` value. The package
+    /// directory is used if a config file is not found.
+    ///
+    /// The `BUILD_WORKSPACE_DIRECTORY` environment variable controls SwiftLint's working directory.
+    ///
+    /// Reference: [https://github.com/realm/SwiftLint/blob/0.50.3/Source/swiftlint/Commands/SwiftLint.swift#L7](
+    /// https://github.com/realm/SwiftLint/blob/0.50.3/Source/swiftlint/Commands/SwiftLint.swift#L7
+    /// )
+    private func environment(
+        context: PluginContext,
+        target: Target
+    ) throws -> [String: String] {
+        let workingDirectory: Path = try target.directory.resolveWorkingDirectory(in: context.package.directory)
+        return ["BUILD_WORKSPACE_DIRECTORY": "\(workingDirectory)"]
+    }
+
+    private func makeCommand(
+        executable: PluginContext.Tool,
+        swiftFiles: [Path],
+        environment: [String: String],
+        pluginWorkDirectory path: Path
+    ) -> [Command] {
+        // Don't lint anything if there are no Swift source files in this target
+        guard !swiftFiles.isEmpty
+        else { return [] }
+        // Outputs the environment to the build log for reference.
+        print("Environment:", environment)
+        let arguments: [String] = [
             "lint",
             "--quiet",
             // We always pass all of the Swift source files in the target to the tool,
             // so we need to ensure that any exclusion rules in the configuration are
             // respected.
             "--force-exclude"
         ]
-
         // Determine whether we need to enable cache or not (for Xcode Cloud we don't)
+        let cacheArguments: [String]
         if ProcessInfo.processInfo.environment["CI_XCODE_CLOUD"] == "TRUE" {
-            arguments.append("--no-cache")
+            cacheArguments = ["--no-cache"]
         } else {
-            arguments.append("--cache-path")
-            arguments.append("\(workingDirectory)")
+            let cachePath: Path = path.appending("Cache")
```

Comment:
Should this be a hidden directory like `.cache`.

Reply:
The `PluginContext` provides the work directory (`path`) to the plugin to use as necessary. Two subdirectories (one named `Cache` and the other named `Output`) are created. The directories do not need to be hidden since they are within the work directory. The cache directory is used for SwiftLint's cache. The output directory, although not used by SwiftLint, is a required argument when creating the `.prebuildCommand(...)`.